src/wp-includes/l10n.php

```diff
@@ -1988,3 +1988,38 @@ function wp_get_word_count_type() {
 
 	return $wp_locale->get_word_count_type();
 }
+
+/**
+ * Parses the `Accept-Language` header to get a list of locales.
+ *
+ * The locales are in the format WordPress expects, so "fr-CH" becomes "fr_CH"
+ * and "fr" becomes "fr_FR". "en" and wildcard values are discarded.
+ * The priority weighting is also discarded.
+ *
+ * @since 6.6.0
+ *
+ * @return string[] Locales list.
+ */
+function get_locales_from_accept_language_header() {
+	$locales = array();
+
+	if ( ! empty( $_SERVER['HTTP_ACCEPT_LANGUAGE'] ) ) {
+		$matches = array();
+		preg_match_all( '((?P<code>[a-z-_A-Z]{2,5})([;q=]+?(?P<prio>0.\d+))?)', $_SERVER['HTTP_ACCEPT_LANGUAGE'], $matches );
+
+		foreach ( $matches['code'] as $code ) {
+			$locale = sanitize_locale_name( str_replace( '-', '_', $code ) );
+
+			if ( 'en' === $code ) {
+				continue;
+			}
+
+			if ( 2 === strlen( $locale ) ) {
+				$locale = $locale . '_' . strtoupper( $locale );
+			}
```

Comment:
Yeah it's an approximation that only works for some locales, not all.

What dou think about using the `available_translations` transient here? It's short-lived, so it's quite possible that 
`get_site_transient` will delete it because it's expired. So maybe we could use `get_site_option( '_site_transient_available_translations' )` directly? Anyway, with this transient we'd have the actual ISO codes that we could use for comparison.

And then this approximation would only be a fallback in case the option doesn't exist.

Alternatively, we could maintain a hardcoded list in this function, but I don't like that either.

Reply:
Another thought:

For two-letter cases we could just add both varians to the array, i.e. `el` and `el_EL`. It would work for `el`, `fr` and `de`, but not for `bn`.

Reply:
I just pushed a version that's way more robust

Reply:
Could the tests include a three-letter case such as Kyrgyz (`kir`) to make sure it works with those too?

Reply:
Good idea. The browser passes this as `ky`, but in WP it's `kir`

Reply:
btw, I am getting the transient this way to avoid remote http requests in `wp_load_translations_early`